log/logger.go

```diff
@@ -1,25 +1,52 @@
 package log
 
 import (
+	"encoding"
+	"encoding/json"
+	"fmt"
 	"io"
 
+	"github.com/bytedance/sonic"
+	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/pkgerrors"
 )
 
+func init() {
+	zerolog.InterfaceMarshalFunc = func(i any) ([]byte, error) {
+		switch v := i.(type) {
+		case json.Marshaler:
+			return sonic.Marshal(i)
+		case encoding.TextMarshaler:
+			return sonic.Marshal(i)
+		case fmt.Stringer:
+			return sonic.Marshal(v.String())
+		default:
+			return sonic.Marshal(i)
+		}
+	}
+}
+
 // ModuleKey defines a module logging key.
 const ModuleKey = "module"
 
 // ContextKey is used to store the logger in the context.
-var ContextKey struct{}
+var ContextKey contextKey
+
+type contextKey struct{}
 
 // Logger is the Cosmos SDK logger interface.
-// It maintains as much backward compatibility with the CometBFT logger as possible.
-// All functionalities of the logger are available through the Impl() method.
+// It extends cosmossdk.io/core/log.Logger to return a child logger.
+// Use cosmossdk.io/core/log.Logger instead in modules.
```

Comment:
we should remove this while we decide on what we will do with all of the `core` work - plus core is not imported to this package